src/main/kotlin/com/kotlindiscord/database/DataModels.kt

```diff
@@ -0,0 +1,42 @@
+package com.kotlindiscord.database
+
+import org.jetbrains.exposed.dao.LongEntity
+import org.jetbrains.exposed.dao.LongEntityClass
+import org.jetbrains.exposed.dao.id.EntityID
+
+class User(id: EntityID<Long>) : LongEntity(id) {
+    companion object : LongEntityClass<User>(Users)
+
+    var discriminator by Users.discriminator
+    var userName by Users.username
+    var avatarUrl by Users.avatarUrl
+    var roles by Role via UserRoles
+}
+
+class Infraction(id: EntityID<Long>) : LongEntity(id) {
+    companion object : LongEntityClass<Infraction>(Infractions)
+
+    var infractor by User referencedOn Infractions.infractor
+    var user by User referencedOn Infractions.user
+    var reason by Infractions.reason
+    var type by Infractions.type
+    var expires by Infractions.expires
+    var created by Infractions.created
+}
+
+class Role(id: EntityID<Long>) : LongEntity(id) {
+    companion object : LongEntityClass<Role>(Roles)
+
+    var name by Roles.name
+    var colour by Roles.colour
+    var holders by User via UserRoles
+}
+
+class AuditLogEntry(id: EntityID<Long>) : LongEntity(id) {
```

Comment:
We're going to skip out on the audit log for the time being, that feature is probably going to need an actual working group haha